### DISCUSSION TITLE
Livewire not responding to Model Broadcast Events

### DISCUSSION THREAD

**Discussion by mikebronner** [livewire/livewire#4831] [1. Help] [answered]
I have set up Laravel Echo using `beyondcode/laravel-websockets` model broadcasting in Laravel 8.x according to the documentation (https://laravel.com/docs/9.x/broadcasting#model-broadcasting). Running the websocket server shows that Livewire successfully registers with the given channel:

```
PFKJ5W3TYTFnv7p5yVRWsBPd: connection id 429277918.590546366 received message: {"event":"pusher:subscribe","data":{"auth":"wttTXkwAPaP8pu2M25MFNv2u:59d0397ca123c03c3e4d1ae00e127ccdff89a04020d3e0d4215d112fcb06da76","channel":"private-customers.3725"}}.
Connection id 429277918.590546366 sending message {"event":"pusher_internal:subscription_succeeded","channel":"private-customers.3725"}
```

This would indicate that the channel authorization is working, and that Laravel is communicating with the Echo client. After making an update on the page, I can see that the websocket server picks up the respective model event:
```
Connection id 578911081.501475891 sending message {"channel":"private-customers.3725","event":"CustomerUpdated","data":"{\"model\":{\"id\":3725, [...]}
```

The listeners are configured as such in my Livewire component:
```php
    public function getListeners(): array
    {
        return [
            "echo-private:customers.{$this->customerId},CustomerUpdated" => "refreshCustomer",
        ];
    }

    public function mount(int $customerId): void
    {
        $this->customerId = $customerId;
    }

    public function refreshCustomer(): void
    {
        logger()->debug("refreshing customer");
        $this->toggleRerender = ! $this->toggleRerender;
    }
```

What I cannot figure out is why `refreshCustomer` never gets called. I believe I followed all the directions for setting up Laravel Echo event listeners correctly (https://laravel-livewire.com/docs/2.x/laravel-echo). All my debugging so far seems to indicate that the events are being sent to Livewire correctly, and that the disconnect is happening somewhere in the Livewire component.

Anyone have any ideas what might be going on here?

**Comment by mikebronner** [comment, marked as answer]
I finally figured it out: Model Broadcast are not real events, so we need to prefix the event name with a period:
```
    public function getListeners(): array
    {
        return [
            "echo-private:customers.{$this->customerId},.CustomerUpdated" => "refreshCustomer",
        ];
    }
```

**Reply by francoism90** [reply]
In addition to this, I needed to use `broadcastAs`.

**Reply by jan-edge** [reply]
@mikebronner Thanks! Why isn't the period prefix mentioned in the livewire documentation?! How the hell did you figure this out? I've wasted a lot of time only to miss a single dot prefix.

**Reply by joshhanley** [reply]
@jan-edge technically, the dot prefix is actual a Laravel Echo thing, so you should be reading both Livewire's and Echo's docs, see here https://laravel.com/docs/10.x/broadcasting#namespaces

But yeah agreed, it probably should also be in Livewire's docs so people don't get tripped up by it. PRs to the docs are welcome https://github.com/livewire/docs

**Reply by edalzell** [reply]
I too wasted so much time on this, thanks @mikebronner. PR'ing into the docs right now

**Reply by viniciusalvess** [reply]
Thanks for sharing this, it helped me to solve my issue. My issue was that I had my event classes on a separate package with my custom namespace for that package and Livewire wasn't finding my events because it assumes the Event classes lives on the default namespace. The "." is to specify the namespace according to the [Laravel Echo](https://laravel.com/docs/10.x/broadcasting#namespaces) docs and as soon as I added the dot followed by my custom namespace it worked like a charm. I'm sharing this because it may help someone else.


```php
...
protected function getListeners(){
    return [
      'refreshHistory' => 'refreshHistory',
      'echo-private:refreshsms,.Vinsystems\\DataPersistence\\Events\\SmsInboundEvent' => 'refreshHistoryListener',
      'echo-private:refreshsms,.Vinsystems\\DataPersistence\\Events\\SmsOutboundEvent' => 'refreshHistoryListener'
    ];
  }
...
```

**Reply by aibarra11** [reply]
thank you! livewire 3 docs don't have that period when defining custom events

**Reply by altayevrim** [reply]
I've lost hours on this one too. The Livewire docs still doesn't have something like that unfortunately. 

My problem was [this one](https://laravel.com/docs/10.x/broadcasting#broadcast-name) exactly 
<img width="731" height="272" alt="image" src="https://github.com/user-attachments/assets/072ec43f-d4d3-49dd-a04e-4bedde555260" />
